test/rules/callable-types/test.ts.lint

```diff
@@ -32,6 +32,16 @@ var fn: {(): void;};
          ~~~~~~~~~ [type % ('() => void')]
 function f(x: { (): void }): void;
                 ~~~~~~~~ [type % ('() => void')]
+function f(x: string | { (): void }): void;
```

Comment:
also add a test where the type is already wrapped in parentheses, e.g. `: string | ({ (): void })`

Reply:
The funny thing is that it still works and doesn't add parens in this case. I guess that it there is already parens, parent.parent doesn't reference the Union, be reference the node with the extra parens.

Reply:
That's right, `parent.parent` is `ts.ParenthesizedType` in that case. That's why I wanted this test case. To verify that we don't add parens if they are already present.